### ISSUE TITLE
Democracy in Action For Naming

### ISSUE THREAD

**Issue by robintibor** [braindecode/braindecode#286]
So right now in quite some models we have:

https://github.com/braindecode/braindecode/blob/4eb0af49824d764d771792931511b449a092f49f/braindecode/models/deep4.py#L38-L39

Let's vote how to change it!

**Comment by robintibor** [comment]
First vote:

`chans` ![autoformatting](https://github.githubassets.com/images/icons/emoji/unicode/1f389.png "Auto formatting")

vs.

`channels` ![pep8](https://github.githubassets.com/images/icons/emoji/unicode/1f680.png "Auto formatting")

**Comment by robintibor** [comment]
Second vote:

`n_in_chan[nel]s, n_out_chan[nel]s` ![1](https://github.githubassets.com/images/icons/emoji/unicode/1f389.png "1")

`n_chan[nels], n_classes` ![2](https://github.githubassets.com/images/icons/emoji/unicode/1f680.png "2")

`in_channels, out_channels` ![3](https://github.githubassets.com/images/icons/emoji/unicode/2764.png "3")

**Comment by robintibor** [comment]
What a marvelous way to vote! @agramfort notice second vote also up

**Comment by agramfort** [comment]
I am fine to have « in » in the name if this also one with « out ». If there’s no ambiguity I don’t see the point

**Comment by agramfort** [comment]
And if we have a pytorch convention and the semantic matches it should be in_channels

**Comment by robintibor** [comment]
keep in mind in the end n_classes loses meaning for regression tasks obviously

**Comment by agramfort** [comment]
Should we rename it to out_channels then ?

**Comment by l-omar-chehab** [comment]
> And if we have a pytorch convention and the semantic matches it should be in_channels

Agreed, it would be nice to have some consistency with Pytorch naming:
-- Convolutional layers in Pytorch use: nn.Conv1d(in_channels, out_channels)
-- Linear layers in Pytorch use: nn.Linear(in_features, out_features)

We could rename in_chans to in_channels for consistency with Pytorch, in braindecode's convolutional models. 

I think it could be worthwhile to keep n_classes as is:
-- this avoids the naming ambiguity of choosing between out_features and out_channels, depending on if the prediction is outputed from a Linear or Conv layer
-- it is more explicit : we understand via n_classes that we are doing a classification task. This is less obvious if we rename n_classes to out_channels, for eg. 

What do you think?

**Comment by agramfort** [comment]
in_channels, out_channels, n_classes would work for me. Honestly everything works as long as it's clear and consistent

**Comment by bruAristimunha** [comment]
Hi!

I'm working on the integration of moabb and braindecode. On the moabb side, we will need to normalize all the variable names in almost all braindecode models. With that, we'll be able to use the full potential of the constructor defined as using a pipeline, doing dynamic searches and stuff like that.

Can I assign this issue to myself and solve the issue directly on braindecode side, @robintibor and @agramfort?

**Comment by bruAristimunha** [comment]
I conducted a search using the stable version of braindecode and extracted the names of all parameters from the initialization of each function. It's not that bad, but it's not good either. 

maybe we need to normalize all the variables and maybe create some tests to make sure this doesn't get out of hand in the future.

Can I follow up with a PR to rename the variables? 

|                        | 0                      | 1                      | 2                     | 3                     | 4         | 5         | 6         | 7         | 8                      | 9                     | 10                    | 11     | 12     |
|:-----------------------|:-----------------------|:-----------------------|:----------------------|:----------------------|:----------|:----------|:----------|:----------|:-----------------------|:----------------------|:----------------------|:-------|:-------|
| in_chans               | ShallowFBCSPNet        | Deep4Net               | EEGNetv1              | EEGNetv4              | HybridNet | EEGResNet | USleep    | TIDNet    |                        |                       |                       |        |        |
| n_classes              | ShallowFBCSPNet        | Deep4Net               | EEGInception          | EEGITNet              | EEGNetv1  | EEGNetv4  | HybridNet | EEGResNet | SleepStagerChambon2018 | SleepStagerBlanco2020 | SleepStagerEldele2021 | USleep | TIDNet |
| input_window_samples   | ShallowFBCSPNet        | Deep4Net               | EEGInception          | EEGITNet              | EEGNetv1  | EEGNetv4  | HybridNet | EEGResNet | TIDNet                 |                       |                       |        |        |
| n_filters_time         | ShallowFBCSPNet        | Deep4Net               |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| filter_time_length     | ShallowFBCSPNet        | Deep4Net               |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| n_filters_spat         | ShallowFBCSPNet        | Deep4Net               |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| pool_time_length       | ShallowFBCSPNet        | Deep4Net               |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| pool_time_stride       | ShallowFBCSPNet        | Deep4Net               |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| final_conv_length      | ShallowFBCSPNet        | Deep4Net               | EEGNetv1              | EEGNetv4              |           |           |           |           |                        |                       |                       |        |        |
| conv_nonlin            | ShallowFBCSPNet        |                        |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| pool_mode              | ShallowFBCSPNet        | EEGNetv1               | EEGNetv4              |                       |           |           |           |           |                        |                       |                       |        |        |
| pool_nonlin            | ShallowFBCSPNet        |                        |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| split_first_layer      | ShallowFBCSPNet        | Deep4Net               | EEGResNet             |                       |           |           |           |           |                        |                       |                       |        |        |
| batch_norm             | ShallowFBCSPNet        | Deep4Net               |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| batch_norm_alpha       | ShallowFBCSPNet        | Deep4Net               | EEGInception          | EEGResNet             |           |           |           |           |                        |                       |                       |        |        |
| drop_prob              | ShallowFBCSPNet        | Deep4Net               | EEGInception          | EEGITNet              | EEGNetv1  | EEGNetv4  | TCN       | TIDNet    |                        |                       |                       |        |        |
| n_filters_2            | Deep4Net               |                        |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| filter_length_2        | Deep4Net               |                        |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| n_filters_3            | Deep4Net               |                        |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| filter_length_3        | Deep4Net               |                        |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| n_filters_4            | Deep4Net               |                        |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| filter_length_4        | Deep4Net               |                        |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| first_conv_nonlin      | Deep4Net               |                        |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| first_pool_mode        | Deep4Net               |                        |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| first_pool_nonlin      | Deep4Net               |                        |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| later_conv_nonlin      | Deep4Net               |                        |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| later_pool_mode        | Deep4Net               |                        |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| later_pool_nonlin      | Deep4Net               |                        |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| stride_before_pool     | Deep4Net               |                        |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| in_channels            | EEGInception           | EEGITNet               |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| sfreq                  | EEGInception           | SleepStagerChambon2018 | SleepStagerBlanco2020 | SleepStagerEldele2021 | USleep    |           |           |           |                        |                       |                       |        |        |
| scales_samples_s       | EEGInception           |                        |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| n_filters              | EEGInception           | TCN                    |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| activation             | EEGInception           |                        |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| depth_multiplier       | EEGInception           |                        |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| pooling_sizes          | EEGInception           |                        |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| second_kernel_size     | EEGNetv1               |                        |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| third_kernel_size      | EEGNetv1               | EEGNetv4               |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| F1                     | EEGNetv4               |                        |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| D                      | EEGNetv4               |                        |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| F2                     | EEGNetv4               |                        |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| kernel_length          | EEGNetv4               |                        |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| final_pool_length      | EEGResNet              |                        |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| n_first_filters        | EEGResNet              |                        |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| n_layers_per_block     | EEGResNet              |                        |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| first_filter_length    | EEGResNet              |                        |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| nonlinearity           | EEGResNet              |                        |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| batch_norm_epsilon     | EEGResNet              |                        |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| conv_weight_init_fn    | EEGResNet              |                        |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| n_in_chans             | TCN                    |                        |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| n_outputs              | TCN                    |                        |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| n_blocks               | TCN                    |                        |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| kernel_size            | TCN                    |                        |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| add_log_softmax        | TCN                    |                        |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| n_channels             | SleepStagerChambon2018 | SleepStagerBlanco2020  |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| n_conv_chs             | SleepStagerChambon2018 |                        |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| time_conv_size_s       | SleepStagerChambon2018 | USleep                 |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| max_pool_size_s        | SleepStagerChambon2018 |                        |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| pad_size_s             | SleepStagerChambon2018 |                        |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| input_size_s           | SleepStagerChambon2018 | SleepStagerBlanco2020  | SleepStagerEldele2021 | USleep                |           |           |           |           |                        |                       |                       |        |        |
| dropout                | SleepStagerChambon2018 | SleepStagerBlanco2020  | SleepStagerEldele2021 |                       |           |           |           |           |                        |                       |                       |        |        |
| apply_batch_norm       | SleepStagerChambon2018 | SleepStagerBlanco2020  |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| return_feats           | SleepStagerChambon2018 | SleepStagerBlanco2020  | SleepStagerEldele2021 |                       |           |           |           |           |                        |                       |                       |        |        |
| n_conv_chans           | SleepStagerBlanco2020  |                        |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| n_groups               | SleepStagerBlanco2020  |                        |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| max_pool_size          | SleepStagerBlanco2020  |                        |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| n_tce                  | SleepStagerEldele2021  |                        |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| d_model                | SleepStagerEldele2021  |                        |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| d_ff                   | SleepStagerEldele2021  |                        |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| n_attn_heads           | SleepStagerEldele2021  |                        |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| after_reduced_cnn_size | SleepStagerEldele2021  |                        |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| depth                  | USleep                 |                        |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| n_time_filters         | USleep                 |                        |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| complexity_factor      | USleep                 |                        |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| with_skip_connection   | USleep                 |                        |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| ensure_odd_conv_size   | USleep                 |                        |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| apply_softmax          | USleep                 |                        |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| s_growth               | TIDNet                 |                        |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| t_filters              | TIDNet                 |                        |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| pooling                | TIDNet                 |                        |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| temp_layers            | TIDNet                 |                        |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| spat_layers            | TIDNet                 |                        |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| temp_span              | TIDNet                 |                        |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| bottleneck             | TIDNet                 |                        |                       |                       |           |           |           |           |                        |                       |                       |        |        |
| summary                | TIDNet                 |                        |                       |                       |           |           |           |           |                        |                       |                       |        |        |

**Comment by agramfort** [comment]
yes it's a good idea to think about this. What renaming are you actually
suggesting?

do we accept any API change without deprecation?

**Comment by robintibor** [comment]
Yes you can do that @bruAristimunha . I would suggest to keep the old names as keyword arguments with default value None and a warning that they will be removed, e.g.
``` python
def __init__(new_name, old_name=None):
    if old_name is not None:
         new_name = old_name
         # some deprecation warning
```

Probably can be made more compact in case of many such renamings with something like
``` python
def __init__(new_name, **kwargs):
    old_new_name_pairs = {old_name:new_name, ...}
    for old_name, new_name in old_new_name_pairs.items():
        if old_name in kwargs:
            setattr(self, new_name, kwargs[old_name])
            # some deprecation warning
```

What do you think @bruAristimunha ?

**Comment by bruAristimunha** [comment]
Hi @agramfort and @robintibor! :)

Definitely follow the warning path. Thank you very much for the code, I will replicate it inside the models!! I like the compacted version, and we could maybe integrate this inside a function and use it in all the models.

I will create a map of all variable names used in the models, and we can discuss the new naming conventions in the pull request. Creating it in parts, so it doesn't get too big, something like first the sleep models and then the BCI models.
